File: solutions/03 annoying blobs/cpp/AnnoyingBlobs/AnnoyingBlobs/DataDisplayer.cpp
#include "DataDisplayer.h"
#include <iostream>
#include "DataBlob.h"

using namespace std;

DataDisplayer::DataDisplayer() :
	_position(0)
{
}

DataDisplayer::~DataDisplayer()
{
}

void DataDisplayer::tick()
{
	int value = DataBlob::getBlob().getDataAtPos(_position);
	cout << "Data @" << _position << "=" << value << endl;
	++_position;
}

File: exercises/03 annoying blobs/cs/AnnoyingBlobs/DataDisplayer.cs
using System;

namespace AnnoyingBlobs
{
    public class DataDisplayer
    {
        public void Tick()
        {
            int value = DataBlob.GetBlob().GetDataAtPos(position);
            Console.WriteLine("Data @" + position + "=" + value);
            position++;
        }

        private int position = 0;
    }
}
